packages/edit-site/src/components/post-fields/index.js

```diff
@@ -191,12 +233,117 @@ function usePostFields( viewType ) {
 	const fields = useMemo(
 		() => [
 			{
-				id: 'featured-image',
+				id: 'featured_media',
 				label: __( 'Featured Image' ),
-				getValue: ( { item } ) => item.featured_media,
+				type: 'image',
```

Comment:
What is this `type` for? This PR provides custom `Edit` and `render` functions and the `type` is not used anywhere I can see.

I see a couple of options:

1. Implement the `image` type for DataForm (with a demo in storybook), like [any other types](https://github.com/WordPress/gutenberg/blob/trunk/packages/dataviews/src/dataform-controls/index.tsx#L1)) we support. The advantage is that we can make `type` a required prop to fields. Then, we'd override the Edit control here to use WordPress-specifics.
2. Implement a custom `feature-image` type that is specific to WordPress, like the [slug PR](https://github.com/WordPress/gutenberg/pull/65196) tries to do.
3. Do not provide any type and just override Edit & render. We'd still need to 1 at some point. It's fine if it's a follow-up.

Reply:
This PR has been opened before `@wordpress/fields` package and the multiple great conversations that we had during these week. I think that it makes sense to follow the second option. I'm going to move this PR as draft, given that it will require some work before another round of review!

Thanks for the help! 🙇